src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -172,6 +172,9 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
         $variables = array_flip($variables);
         $mergedParams = array_replace($defaults, $this->context->getParameters(), $parameters);
 
+        // force string for an object. See https://bugs.php.net/bug.php?id=66966
+        $parameters = array_map(static function ($param) { return \is_object($param) ? (string) $param : $param; }, $parameters);
```

Comment:
I wonder if you should perform this operation on the `$extra` array instead, which holds only those parameters that will be funneled into `http_build_query()`.